### PR TITLE
ARROW-8605: [R] Add brotli to Windows R build

### PR DESCRIPTION
**Pull request by wjones127**
Adds brotli compression to Windows R build.

In addition, Windows CI R Check jobs will now dump the test output.

Added brotli to rtools-packages here: https://github.com/r-windows/rtools-packages/pull/232

### CODE REVIEW

File: ci/scripts/r_windows_build.sh [hunk 1]
```diff
@@ -87,7 +87,7 @@ if [ -d mingw64/lib/ ]; then
   # These may be from https://dl.bintray.com/rtools/backports/
   cp $MSYS_LIB_DIR/mingw64/lib/lib{thrift,snappy}.a $DST_DIR/${RWINLIB_LIB_DIR}/x64
   # These are from https://dl.bintray.com/rtools/mingw{32,64}/
-  cp $MSYS_LIB_DIR/mingw64/lib/lib{zstd,lz4,crypto,utf8proc,re2,aws*}.a $DST_DIR/lib/x64
+  cp $MSYS_LIB_DIR/mingw64/lib/lib{zstd,lz4,brotlicommon,brotlidec,brotlienc,crypto,utf8proc,re2,aws*}.a $DST_DIR/lib/x64
```
Comment (pitrou) [inline ci/scripts/r_windows_build.sh:90]: Perhaps simply `brotli*` here and below?

File: r/configure.win [hunk 2]
```diff
@@ -52,7 +53,7 @@ function configure_release() {
   # NOTE: If you make changes to the libraries below, you should also change
   # ci/scripts/r_windows_build.sh and ci/scripts/PKGBUILD
   PKG_CFLAGS="-I${RWINLIB}/include -DARROW_STATIC -DPARQUET_STATIC -DARROW_DS_STATIC -DARROW_R_WITH_ARROW -DARROW_R_WITH_PARQUET -DARROW_R_WITH_DATASET -DARROW_R_WITH_JSON"
-  PKG_LIBS="-L${RWINLIB}/lib"'$(subst gcc,,$(COMPILED_BY))$(R_ARCH) '"-L${RWINLIB}/lib"'$(R_ARCH)$(CRT) '"-lparquet -larrow_dataset -larrow -larrow_bundled_dependencies -lutf8proc -lthrift -lsnappy -lz -lzstd -llz4 -lole32 ${MIMALLOC_LIBS} ${OPENSSL_LIBS}"
+  PKG_LIBS="-L${RWINLIB}/lib"'$(subst gcc,,$(COMPILED_BY))$(R_ARCH) '"-L${RWINLIB}/lib"'$(R_ARCH)$(CRT) '"-lparquet -larrow_dataset -larrow -larrow_bundled_dependencies -lutf8proc -lthrift -lsnappy -lz -lzstd -llz4 ${BROTLI_LIBS} -lole32 ${MIMALLOC_LIBS} ${OPENSSL_LIBS}"
```
Comment (pitrou) [inline r/configure.win:56]: While we are at it, can this easily (and the `PKG_CFLAGS` variable above) be wrapped over multiple lines for better readability?

File: r/tests/testthat/test-compressed.R [hunk 3]
```diff
@@ -27,6 +27,7 @@ if (identical(Sys.getenv("APPVEYOR"), "True")) {
   test_that("Compression codecs are included in the Windows build", {
```
Comment (pitrou) [inline r/tests/testthat/test-compressed.R:27]: We don't have any R builds on AppVeyor, do we? The condition should be that the test is running on Windows instead.

File: r/configure.win (hunk 2 above)
Comment (wjones127) [reply r/configure.win:56]: Yes, I'll wrap them to 90 chars. Also doing that for the aws libs list near the top.

File: r/tests/testthat/test-compressed.R (hunk 3 above)
Comment (wjones127) [reply r/tests/testthat/test-compressed.R:27]: Good catch! these tests probably haven't been running. I don't see a good way to check if a particular test ran in the CI. I'll double check on my local machine tomorrow.

Comment (wjones127) [reply r/tests/testthat/test-compressed.R]: I've fixed the condition and moved it into a `skip_if()` call, which allows us to see in CI output whether it was skipped.

File: ci/scripts/r_windows_build.sh [hunk 4]
```diff
@@ -97,15 +97,15 @@ if [ -d mingw32/lib/ ]; then
   mkdir -p $DST_DIR/lib/i386
   mv mingw32/lib/*.a $DST_DIR/${RWINLIB_LIB_DIR}/i386
   cp $MSYS_LIB_DIR/mingw32/lib/lib{thrift,snappy}.a $DST_DIR/${RWINLIB_LIB_DIR}/i386
-  cp $MSYS_LIB_DIR/mingw32/lib/lib{zstd,lz4,crypto,utf8proc,re2,aws*}.a $DST_DIR/lib/i386
+  cp $MSYS_LIB_DIR/mingw32/lib/lib{zstd,lz4,brotlicommon,brotlidec,brotlienc,crypto,utf8proc,re2,aws*}.a $DST_DIR/lib/i386
```
Comment (pitrou) [inline ci/scripts/r_windows_build.sh:100]: Is there a reason for not using `brotli*` here and below?

**Comment by pitrou** [comment]
@github-actions crossbow submit -g r

**Comment by wjones127** [comment]
@github-actions crossbow submit -g r

File: ci/scripts/r_windows_build.sh (hunk 4 above)
Comment (wjones127) [reply ci/scripts/r_windows_build.sh]: No reason not to. Fixed now.

**Review by pitrou — APPROVED** [review]

**Pull request closed without merging**